RIOT-OS/RIOT
# cpu/stm32: Fix clock tree

### Contribution description

This fixes the kconfig clock tree for `pyboard` and `msbiot`.
The kconfig modelling needs to be done in the board as there is a dependency on the USB dev.
The clock tree header file is also adapted to support the 16 MHz crystal from the `msbiot`.
The boards are removed from the blocklist as a result.

### Testing procedure
Green murdock for the kconfig changes, maybe try to check the clockout for the msbiot to see that it is actually at 180MHz.

### Issues/PRs references

Maybe related to #14975
